Orbit.h

```diff
@@ -129,49 +126,129 @@ enum class Color : uint32_t {
 
 namespace orbit_api {
 
+constexpr uint8_t kVersion = 1;
+
+enum EventType : uint8_t {
+  kNone = 0,
+  kScopeStart = 1,
+  kScopeStop = 2,
+  kScopeStartAsync = 3,
+  kScopeStopAsync = 4,
+  kTrackInt = 5,
+  kTrackInt64 = 6,
+  kTrackUint = 7,
+  kTrackUint64 = 8,
+  kTrackFloat = 9,
+  kTrackDouble = 10,
+};
+
+constexpr size_t kMaxEventStringSize = 30;
+struct Event {
+  uint8_t version;                 // 1
+  uint8_t type;                    // 1
+  char name[kMaxEventStringSize];  // 30
+  uint64_t value;                  // 8
+  orbit::Color color;              // 4
+  uint32_t id;                     // 4
+};
+
+union EncodedEvent {
+  EncodedEvent(orbit_api::EventType type, const char* name = nullptr, uint64_t value = 0,
+               orbit::Color color = orbit::Color::kAuto, uint32_t id = 0) {
+    static_assert(sizeof(EncodedEvent) == 48, "EncodedEvent should be 48 bytes.");
+    memset(this, 0, sizeof(*this));
```

Comment:
Is this to clear `name` (the other values are set)? Can we explicitly clear only `name` then?

Reply:
Yep, done!